homeassistant/components/http/cors.py

```diff
@@ -50,7 +52,11 @@ def allow_cors(route, methods):
                 route = route.resource
             if route in cors_added:
```

Comment:
This is the key. We need to populate this set with cors_allowed views before doing this step.